### PR TITLE
Address PR feedback on https://github.com/dotnet/roslyn/pull/74132

### PR DESCRIPTION
**Pull request by 333fred**

### CODE REVIEW

**Comment by 333fred** [comment]
@jcouv @dotnet/roslyn-compiler for review. Followup from non-blocking comments on https://github.com/dotnet/roslyn/pull/74132.

File: src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs [hunk 1]
```diff
@@ -289,64 +308,64 @@ private sealed class UncommonFields
...
-            var retVal = new UncommonFields();
-            if (!_packedFlags.IsObsoleteAttributePopulated)
-            {
-                retVal._lazyObsoleteAttributeData = ObsoleteAttributeData.Uninitialized;
-            }
+            var retVal = _uncommonFields;
+            return retVal ?? InterlockedOperations.Initialize(ref _uncommonFields, createUncommonFields());
 
-            if (!_packedFlags.IsUnmanagedCallersOnlyAttributePopulated)
+            UncommonFields createUncommonFields()
```
Comment (jcouv) [inline src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs:316]: static?

File: src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs [hunk 2]
```diff
@@ -271,8 +271,27 @@ public bool InitializeIsUnscopedRef(bool value)
         }
 
         /// <summary>
-        /// Holds infrequently accessed fields. See <seealso cref="_uncommonFields"/> for an explanation.
+        /// This type is used hold lazily-initialized fields that many method will not need. We avoid creating it unless one the fields is needed;
```
Comment (jcouv) [inline src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs:274]: >used hold [](http://example.com/codeflow?start=25&length=9)

"used hold" => "used to hold"

Comment (jcouv) [inline src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs:274]: >many method [](http://example.com/codeflow?start=66&length=11)

"many method" => "many methods"

Comment (jcouv) [inline src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs:274]: >one the [](http://example.com/codeflow?start=121&length=7)

"one the" => "one of the"

File: src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEPropertySymbol.cs [hunk 3]
```diff
@@ -143,6 +143,28 @@ public void SetCustomAttributesPopulated()
             public readonly bool IsCustomAttributesPopulated => (_bits & IsCustomAttributesPopulatedBit) != 0;
         }
 
+        /// <summary>
+        /// This type is used hold lazily-initialized fields that many properties will not need. We avoid creating it unless one the fields is needed;
```
Comment (jcouv) [inline src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEPropertySymbol.cs:147]: >used hold [](http://example.com/codeflow?start=25&length=9)

"used hold" => "used to hold"
"one the" => "one of the"

**Review by jcouv — COMMENTED** [review]
Done with review pass (iteration 1)

**Comment by 333fred** [comment]
@jcouv @dotnet/roslyn-compiler for reviews

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 2)

**Comment by 333fred** [comment]
@dotnet/roslyn-compiler for a second review.

**Comment by 333fred** [comment]
@dotnet/roslyn-compiler for a second review please

**Review by cston — APPROVED** [review]

**Pull request merged**